server/users.ts

```diff
@@ -407,48 +377,102 @@ class Connection {
 	onDisconnect() {
 		connections.delete(this.id);
 		if (this.user) this.user.onDisconnect(this);
-		this.user = /** @type {any} */ (null);
+		this.user = null as any as User;
```

Comment:
```suggestion
		this.user = null!;
```

Reply:
I think I prefer the explicit cast here because it makes it clearer you're asserting it as a type it's not

Reply:
I feel like `null!` is obviously asserting a type it's not, though.

Reply:
This is approximately the "implicitly unwrapped optional" feature in Swift – it's a property asserted non-null because it's only null during initialization and after destruction.

If you wanted to be any more explicit, I would comment that it can be null in those cases in a doc comment on the property. Doing it here isn't adding any safety/readability to my eyes.

Reply:
sure, I'm not too strongly opinionated on this